src/makeCalculator.js

```diff
@@ -4,7 +4,41 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const initialValue = 0;
+
+  const config = {
+    result: initialValue,
+
+    add(value) {
+      this.result += value;
+    },
+
+    subtract(value) {
+      this.result -= value;
+    },
+
+    multiply(value) {
+      this.result *= value;
+    },
+
+    divide(value) {
+      this.result /= value;
+    },
+
+    reset() {
+      this.result = initialValue;
+
+      return this;
+    },
```

Comment:
The `reset` method correctly resets the `result` to the initial value and returns `this` for chaining.